src/west/app/main.py

```diff
@@ -479,6 +491,15 @@ def run_command(self, argv, early_args):
         # If we're running an extension, instantiate it from its
         # spec and re-parse arguments before running.
 
+        # Replace alias command(s) if set
+        while early_args.command_name in self.aliases:
+            new_args = shlex.split(self.aliases[early_args.command_name])
+            # Make sure we don't end up in an infinite loop
+            del self.aliases[early_args.command_name]
```

Comment:
To be on the safe side, can you please work on a local clone of `self.aliases` here and leave `self.aliases` immutable? I don't think there is much concurrency in `west` yet but let's not leave "landmines" for anyone trying to add some in the future.